designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,963 @@
+}];
+```
+
+Here, the directory `web_modules` will be ignored as if it were defined in an `.eslintignore` file. The `web_modules` directory will be excluded from the glob pattern used to determine which files ESLint will run against.
+
+For backwards compatibility, users could create a config like this:
+
+```js
+const fs = require("fs");
+
+module.exports = [
+    {
+        ignores: fs.readFileSync(".eslintignore", "utf8").split("\n")
+    }
+];
+```
+
+### Replacing `--ext`
+
+The `--ext` flag is currently used to pass in one or more file extensions that ESLint should search for when a directory without a glob pattern is passed on the command line, such as:
+
+```bash
+eslint src/ --ext .js,.jsx
+```
+
+This curently searches all subdirectories of `src/` for files with extensions matching `.js` or `.jsx`.
+
+This proposal removes `--ext` by allowing the same information to be passed in a config. For example, the following config achieves the same result:
+
+```js
+const fs = require("fs");
+
+module.exports = {
+    files: ["*.js", "*.jsx"],
+};
+```
+
+ESLint could then be run with this command:
+
+```bash
+eslint src/
+```
+
+When evaluating the `files` array in the config, ESLint will end up searching for `src/**/*.js` and `src/**/*.jsx`. (More information about file resolution is included later this proposal.)
+
+Additionally, ESLint can be run without specifying anything on the command line, relying just on what's in `eslint.config.js` to determine what to lint:
+
+```bash
+eslint
+```
+
+This will go into the `eslint.config.js` and use all of the `files` glob patterns to determine which files to lint.
+
+### Replacing `--rulesdir`
+
+In order to recreate the functionality of `--rulesdir`, a user would need to create a new entry in `plugins` and then specify the rules from a directory. This can be accomplished using the [`requireindex`](https://npmjs.com/package/requireindex) npm package:
+
+```js
+const requireIndex = require("requireindex");
+
+module.exports = {
+    plugins: {
+        custom: {
+            rules: requireIndex("./custom-rules")
+        }
+    },
+    rules: {
+        "custom/my-rule": "error"
+    }
+};
+```
+
+The `requireIndex()` method returns an object where the keys are the rule IDs (based on the filenames found in the directory) and the values are the rule objects. Unlike today, rules loaded from a local directory must have a namespace just like plugin rules (`custom` in this example).
+
+### Function Configs
+
+Some users may need information from ESLint to determine the correct configuration to use. To allow for that, `module.exports` may also be a function that returns an object, such as:
+
+```js
+module.exports = (context) => {
+
+    // do something
+
+    return {
+        files: ["*.js"],
+        rules: {
+            "semi": ["error", "always"]
+        }
+    };
+
+};
+```
+
+The `context` object has the following members:
+
+* `name` - the name of the application being used
```

Comment:
Not clear to me what that "name of the application" is. ESLint? Or the config name?

Reply:
For ESLint it would be "eslint". If, for example, Standard decided to support some configuration, they could change it to "standard".